### ISSUE TITLE
tidal-dl not working - wrong api key

### ISSUE THREAD

**Issue by GinjuGG** [RandomNinjaAtk/arr-scripts#11]
tidal-dl runs into an issue but won't let me re-authenticate 
it restarts the scripts everytime it tries to download something

```
2023-07-13 18:57:13 :: Audio :: 1.3 :: TIDAL :: Verifying tidal-dl configuration
2023-07-13 18:57:13 :: Audio :: 1.3 :: TIDAL :: No default config found, importing default config "tidal.json"
2023-07-13 18:57:13 :: Audio :: 1.3 :: TIDAL :: ERROR :: Loading client for required authentication, please authenticate, then exit the client...

 /$$$$$$$$ /$$       /$$           /$$               /$$ /$$
|__  $$__/|__/      | $$          | $$              | $$| $$
   | $$    /$$  /$$$$$$$  /$$$$$$ | $$          /$$$$$$$| $$
   | $$   | $$ /$$__  $$ |____  $$| $$ /$$$$$$ /$$__  $$| $$
   | $$   | $$| $$  | $$  /$$$$$$$| $$|______/| $$  | $$| $$
   | $$   | $$| $$  | $$ /$$__  $$| $$        | $$  | $$| $$
   | $$   | $$|  $$$$$$$|  $$$$$$$| $$        |  $$$$$$$| $$
   |__/   |__/ \_______/ \_______/|__/         \_______/|__/
   
       https://github.com/yaronzz/Tidal-Media-Downloader 
       
                        2022.10.31.1

+-------------------------------+--------------------------------------------------------------------------+
| SETTINGS                      | VALUE                                                                    |
+-------------------------------+--------------------------------------------------------------------------+
| Settings path                 | /run/s6-rc:s6-rc-init:kOcpII/servicedirs/custom-svc-Audio/.tidal-dl.json |
| Download path                 | /sabnzbd_finished/incomplete                                             |
| Album folder format           | {ArtistName}/{Flag} {AlbumTitle} [{AlbumID}] [{AlbumYear}]               |
| Playlist folder format        | Playlist/{PlaylistName} [{PlaylistUUID}]                                 |
| Track file format             | {TrackNumber} - {ArtistName} - {TrackTitle}{ExplicitFlag}                |
| Video file format             | {VideoNumber} - {ArtistName} - {VideoTitle}{ExplicitFlag}                |
| Audio quality                 | AudioQuality.HiFi                                                        |
| Video quality                 | VideoQuality.P360                                                        |
| Use playlist folder           | True                                                                     |
| Check exist                   | True                                                                     |
| Show progress                 | True                                                                     |
| Show Track Info               | True                                                                     |
| Save AlbumInfo.txt            | False                                                                    |
| Save covers                   | True                                                                     |
| Include singles & EPs         | True                                                                     |
| Language                      | English                                                                  |
| Save timed lyrics (.lrc file) | False                                                                    |
| Multi thread download         | False                                                                    |
| APIKey support                | [0]Normal/High/HiFi(No Master)                                           |
| Use Download Delay            | True                                                                     |
+-------------------------------+--------------------------------------------------------------------------+
[INFO] Current APIKeys: 0 Fire TV-Normal/High/HiFi(No Master)
[INFO] Current Version: 1.0.1
-------------API-KEYS---------------
+-------+-------+--------------+-----------------------------+
| Index | Valid | Platform     | Formats                     |
+-------+-------+--------------+-----------------------------+
| 0     | False | Fire TV      | Normal/High/HiFi(No Master) |
| 1     | True  | Fire TV      | Master-Only(Else Error)     |
| 2     | False | Android TV   | Normal/High/HiFi(No Master) |
| 3     | False | TV           | Normal/High/HiFi/Master     |
| 4     | True  | Android Auto | Normal/High/HiFi/Master     |
+-------+-------+--------------+-----------------------------+
APIKEY index:Traceback (most recent call last):
  File "/usr/bin/tidal-dl", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.11/site-packages/tidal_dl/__init__.py", line 83, in main
    changeApiKey()
  File "/usr/lib/python3.11/site-packages/tidal_dl/events.py", line 196, in changeApiKey
    index = int(Printf.enterLimit("APIKEY index:", LANG.select.MSG_INPUT_ERR, apiKey.getLimitIndexs()))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/tidal_dl/printf.py", line 159, in enterLimit
    ret = aigpy.cmd.inputLimit(aigpy.cmd.yellow(string), limit)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/aigpy/cmdHelper.py", line 88, in inputLimit
    string = input()
             ^^^^^^^
EOFError: EOF when reading a line
```

**Comment by RandomNinjaAtk** [comment]
Update to the latest and see if it resolves the issue: 
https://github.com/RandomNinjaAtk/arr-scripts/tree/main/lidarr#updating

**Comment by GinjuGG** [comment]
```
2023-07-13 20:11:43 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:45 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:47 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:49 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:51 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:53 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:55 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:57 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:11:59 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:12:01 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
2023-07-13 20:12:03 :: Audio :: 1.4 :: STATUS :: TIDAL-DL :: BUSY :: Pausing/waiting for all active tidal-dl tasks to end...
```

Now I get this in Audio.txt

Also PlexNotify is somehow weird. It sends notifications even if I disable it completely in Lidarr->Connect

```
2023-07-13 20:12:43 :: PlexNotify :: 1.0.8 :: Plex Scan notification sent! (3 :: custom-svc-PlexNotify.bash)
2023-07-13 20:12:44 :: PlexNotify :: 1.0.8 :: Plex Connection Established, version: 1.32.4.7195-7c8f9d3b6
2023-07-13 20:12:45 :: PlexNotify :: 1.0.8 :: Plex Scan notification sent! (3 :: custom-svc-PlexNotify.bash)
2023-07-13 20:12:47 :: PlexNotify :: 1.0.8 :: Plex Connection Established, version: 1.32.4.7195-7c8f9d3b6
2023-07-13 20:12:47 :: PlexNotify :: 1.0.8 :: Plex Scan notification sent! (3 :: custom-svc-PlexNotify.bash)
2023-07-13 20:12:49 :: PlexNotify :: 1.0.8 :: Plex Connection Established, version: 1.32.4.7195-7c8f9d3b6
2023-07-13 20:12:50 :: PlexNotify :: 1.0.8 :: Plex Scan notification sent! (3 :: custom-svc-PlexNotify.bash)
2023-07-13 20:12:51 :: PlexNotify :: 1.0.8 :: Plex Connection Established, version: 1.32.4.7195-7c8f9d3b6
```

**Comment by RandomNinjaAtk** [comment]
Open a separate issue for that, you'll need to provide more logs. 

But sounds like something didn't get configured correctly. Might be user error, but not enough to know what is happening

**Comment by GinjuGG** [comment]
I used my old folders where the lidarr-extended container was placed in with the old config and caches and switched to lscr.io/linuxserver/lidarr:developinstalling your scripts following your guide. That doesnt seem to work because tidal couldnt find the right configuration (?) I don't exactly know how the script works.

However, I created a backup, deleted all contents in my Lidarr folder, starting completely fresh, installed lscr.io/linuxserver/lidarr:develop and your scripts, restored backup and now it went back to function as it should without losing any data.

That also fixed the 2s loop on PlexNotify (and MetadataPostProcess)
